src/Core/Util/Entity/EntityNameDuplicator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util\Entity;
+
+use Language;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class EntityNameDuplicator
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        StringModifierInterface $stringModifier
+    ) {
+        $this->translator = $translator;
+        $this->stringModifier = $stringModifier;
+    }
+
+    /**
+     * Adds a "copy" word to localized entity names
+     *
+     * @param array<int, string> $currentLocalizedNames Array with localized names, IdLang => Name
+     * @param int $maxLength Max length of a new name, will be cut if it doesn't fit with the new "copy" word.
+     *
+     * @return array<int, string>
+     */
+    public function getNewLocalizedNames(array $currentLocalizedNames, int $maxLength): array
+    {
+        $newLocalizedNames = [];
+        foreach ($currentLocalizedNames as $langId => $currentName) {
+            $langId = (int) $langId;
+
+            // Add copy word to the string
+            $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', Language::getLocaleById($langId));
```

Comment:
For now the hard-coded message and namespace is fine since it's only used in the Catalog domain, but maybe we'll have to make it configurable in the future For now it's enough